stdlib/src/mmap.rs

```diff
@@ -0,0 +1,1037 @@
+            let size = self.inner_size();
+            let start = options.start.unwrap_or(pos);
+            let end = options.end.unwrap_or(size);
+
+            let size = size.try_into().unwrap();
+            (saturate_index(start, size), saturate_index(end, size))
+        }
+
+        #[pymethod]
+        fn find(&self, options: FindOptions, vm: &VirtualMachine) -> PyResult<PyIntRef> {
+            let (start, end) = self.get_find_range(options.clone());
+
+            let sub = &options.sub;
+
+            if sub.is_empty() {
+                return Ok(PyInt::from(0isize).into_ref(vm));
+            }
+            let pos = match self.check_valid(vm)?.deref().as_ref().unwrap() {
+                MmapObj::Read(mmap) => mmap[start..end]
+                    .windows(sub.len())
+                    .position(|window| window == sub),
+                MmapObj::Write(mmap) => mmap[start..end]
+                    .windows(sub.len())
+                    .position(|window| window == sub),
+            };
+
+            if let Some(i) = pos {
+                return Ok(PyInt::from(start + i).into_ref(vm));
+            }
+
+            Ok(PyInt::from(-1isize).into_ref(vm))
+        }
+
+        #[pymethod]
+        fn rfind(&self, options: FindOptions, vm: &VirtualMachine) -> PyResult<PyIntRef> {
+            let (start, end) = self.get_find_range(options.clone());
+
+            let sub = &options.sub;
+            if sub.is_empty() {
+                return Ok(PyInt::from(0usize).into_ref(vm));
+            }
+
+            let pos = match self.check_valid(vm)?.deref().as_ref().unwrap() {
+                MmapObj::Read(mmap) => mmap[start..end]
+                    .windows(sub.len())
+                    .rposition(|window| window == sub),
+                MmapObj::Write(mmap) => mmap[start..end]
+                    .windows(sub.len())
+                    .rposition(|window| window == sub),
+            };
+
+            if let Some(i) = pos {
+                return Ok(PyInt::from(start + i).into_ref(vm));
+            }
+
+            Ok(PyInt::from(-1).into_ref(vm))
+        }
+
+        #[pymethod]
+        fn flush(&self, options: FlushOptions, vm: &VirtualMachine) -> PyResult<()> {
+            let offset = options.offset.unwrap_or(0);
+            let size = options.size.unwrap_or_else(|| self.inner_size());
+
+            if size < 0 || offset < 0 || self.inner_size() - offset < size {
+                return Err(vm.new_value_error("flush values out of range".to_owned()));
+            }
+
+            let size = size as usize;
+            let offset = offset as usize;
+
+            if self.access == AccessMode::Read || self.access == AccessMode::Copy {
+                return Ok(());
+            }
+
+            match self.check_valid(vm)?.deref().as_ref().unwrap() {
+                MmapObj::Read(_mmap) => {}
+                MmapObj::Write(mmap) => {
+                    mmap.flush_range(offset as usize, size as usize)
```

Comment:
```suggestion
                    mmap.flush_range(offset, size)
```
now we have casting above